inputters/sil.lua

```diff
@@ -201,8 +201,17 @@ function inputter:parse (doc)
     return SU.error("Unable to parse input document to an AST tree")
   end
   resetCache()
-  local ast = massage_ast(tree, doc)
-  return ast
+  local top = massage_ast(tree, doc)
+  local ast = {}
+  for _, leaf in ipairs(top) do
+    if leaf.lno and leaf.command then
+      table.insert(ast, leaf)
+    end
+  end
+  if #ast == 0 then
+    SU.error("This isn't a SILE document!")
+  end
+  return #ast == 1 and ast[1] or { command = "document", ast }
```

Comment:
Franly, I have no idea what this tries to do.

Reply:
Not as a root document no, but in the context of an imported fragment, I could see asking the parser to parse a fragment with no top level element, `foo \code{bar} baz` being a viable SIL fragment (not document, just valid syntax for a fragment to include into another one.

Reply:
I don't follow the rationale fully for this. But admittedly, being sick this week, my brain might be malfunctioning.

Reply:
I still think this use case makes sense but it needs a test and it shouldn't be mashed in the other unrelated change. I'll back it out for now.

Reply:
Just re-reading myself, the above comment might sound ironic but that wasn't intended. I'm really sick, functioning at reduced brain level.